Merge Conflict/Assets/Scripts/UiManager.cs

```diff
@@ -0,0 +1,186 @@
+        MAINMENU.enabled = false;
+        SETTINGS.enabled = false;
+        LEVEL.enabled = false;
+        UPGRADE.enabled = false;
+        ELEMENTS.enabled = false;
+
+        //setup eventlisteners for all buttons
+        setupButtonListener(buttonOpenMainmenu);
+        setupButtonListener(buttonCloseMainmenu);
+        setupButtonListener(buttonSettings);
+        setupButtonListener(buttonLevel);
+        setupButtonListener(buttonUpgrade);
+        setupButtonListener(buttonElements);
+        setupButtonListener(buttonExitGame);
+    }
+
+    private void setupButtonListener(Button button)
+    {
+        if (button != null)
+        {
+            button.onClick.AddListener(() => handleButtonClick(button.name));
+        }
+        else
+        {
+            Debug.LogError($"Button with name {button} not found. Please check if the button exists and ist linkt to the script UiManager!");
+        }
+    }
+
+    private void handleButtonClick(string clickedButton)
+    {
+        if (clickedButton == EXIT_GAME)
+        {
+#if UNITY_EDITOR
+                UnityEditor.EditorApplication.isPlaying = false;
+#else
+            Application.Quit();
+#endif
+            return;
+        }
+
+        screenSwitcher(clickedButton);
+    }
+
+    private void screenSwitcher(string requestedScreen)
+    {
+        if (currentOpenedScreen == NO_OPEN_SCREEN)
+        {
+            currentOpenedScreen = MAINMENU;
+        }
+
+        //close current opened screen
+        currentOpenedScreen.enabled = false;
+
+        //open requested screen with usage of the mapping
+        KeyValuePair<string, string> screenname = buttonRelationScreen.Find(pair => pair.Key == requestedScreen);
+
+        switch (screenname.Value)
+        {
+            case "Mainmenu":
+                MAINMENU.enabled = true;
+                PLAYFIELD.enabled = false;
+                currentOpenedScreen = MAINMENU;
+                menuVisibility = OPENED;
+                break;
+
+            case "Settings":
+                SETTINGS.enabled = true;
+                currentOpenedScreen = SETTINGS;
+                break;
+
+            case "Level":
+                LEVEL.enabled = true;
+                currentOpenedScreen = LEVEL;
+                break;
+
+            case "Upgrade":
+                UPGRADE.enabled = true;
+                currentOpenedScreen = UPGRADE;
+                break;
+
+            case "Elements":
+                ELEMENTS.enabled = true;
+                currentOpenedScreen = ELEMENTS;
+                break;
+
+            case "CloseMenu":
+                MAINMENU.enabled = false;
+                PLAYFIELD.enabled = true;
+                currentOpenedScreen = null;
+                menuVisibility = CLOSED;
+                break;
+
+            default:
+                Debug.LogWarning("No Screen to open with the name: " + screenname.Value);
+                break;
+        }
+    }
+
+    public bool getMenuVisibility()
```

Comment:
refactor by using this: "public int Dollar { get; private set };" for menuVisibility